stdlib/Test/src/Test.jl

```diff
@@ -704,7 +725,16 @@ function do_test_throws(result::ExecutionResult, orig_expr, extype)
             orig_expr isa Expr &&
             orig_expr.head in (:call, :macrocall) &&
             orig_expr.args[1] in MACROEXPAND_LIKE
-        if isa(extype, Type)
+        if isa(extype, AbstractString) || isa(extype, AbstractPattern)
+            exc = '"' * escape_string(sprint(showerror, exc)) * '"'
+            if isa(extype, AbstractString)
```

Comment:
Can we use the test_warn matchers here for consistency?

Reply:
Done